lib/msal-browser/src/config/Configuration.ts

```diff
@@ -125,9 +127,10 @@ export function buildConfiguration({ auth: userInputAuth, cache: userInputCache,
         ...DEFAULT_SYSTEM_OPTIONS,
         loggerOptions: DEFAULT_LOGGER_OPTIONS,
         networkClient: BrowserUtils.getBrowserNetworkClient(),
-        windowHashTimeout: DEFAULT_POPUP_TIMEOUT_MS,
-        iframeHashTimeout: DEFAULT_IFRAME_TIMEOUT_MS,
-        loadFrameTimeout: BrowserUtils.detectIEOrEdge() ? 500 : 0,
+        // If loadFrameTimeout is provided, use that as default.
+        windowHashTimeout: (userInputSystem && userInputSystem.loadFrameTimeout) || DEFAULT_POPUP_TIMEOUT_MS,
+        iframeHashTimeout: (userInputSystem && userInputSystem.loadFrameTimeout) || DEFAULT_IFRAME_TIMEOUT_MS,
```

Comment:
An argument could be made that if `loadFrameTimeout` is set to a really low number (due to how its used in v2) that is smaller than the existing defaults, we should use the defaults (to prevent unintentional timeouts).

Reply:
At a minimum there should probably be a log message.

Reply:
I see the default value as a way to guarantee that unintentional timeouts would not happen, and the user can fine-tune this for their application. I am ok with a log message but would probably vote against overwriting smaller values.

Reply:
Sorry I misread this code. I would prefer not to overwrite user values, but in this case behavior of the parameter is changing, and it may break users using both `loadFrameTimeout` and `windowHashTimeout/iframeHashTimeout`.

Reply:
Adding log messages for now, will look at overwriting user values if we get feedback accordingly.